typegate/tests/type_nodes/array_of_optional.py

```diff
@@ -0,0 +1,49 @@
+from typegraph import policies
+from typegraph import t
+from typegraph import TypeGraph
+from typegraph.runtimes.deno import PredefinedFunMat
+
+with TypeGraph("array_optional") as g:
+    nested = t.struct(
+        {
+            "a": t.string(),
+            "b": t.integer(),
+            "c": t.struct({"c1": t.string(), "inner": t.array(t.string().optional())}),
+        }
+    )
+
+    rec = t.struct(
+        {
+            "struct_array": t.array(nested.optional()).optional(),
+            "string_array": t.array(t.string().optional()).optional(),
+            "integer_array": t.array(t.integer().optional()).optional(),
+            # TODO:
+            # should we support null value for enum/union/either arrays ?
```

Comment:
Not too sure if we should, but I think it's in the JSON Schema so maybe it'd be a good idea.

A naive implementation in the typegraph could be:

```python
class optional(typedef):
    def __attrs_post_init__(self):
        super().__attrs_post_init__()

        of = self.of

        if isinstance(of, NodeProxy):
            of = of.get()

        if of._enum is not None:
            # an optional type with an enum requires the enum to have the value
            # `null` as variant
            #
            # optional-enum(red, blue) -> optiona-enum(red, blue, null)
            object.__setattr__(self, "of", of.replace(enum=(*of._enum, None)))
```